Libraries/LibWeb/HTML/Navigator.cpp

```diff
@@ -148,4 +149,35 @@ GC::Ref<MediaCapabilitiesAPI::MediaCapabilities> Navigator::media_capabilities()
     return *m_media_capabilities;
 }
 
+// https://w3c.github.io/battery/#the-getbattery-method
+GC::Ref<WebIDL::Promise> Navigator::get_battery()
+{
+    auto& realm = this->realm();
+
+    // 1. If this.[[BatteryPromise]] is null, then set it to a new promise in this's relevant realm.
+    if (!m_battery_promise) {
+        // FIXME: Don't reject promise here; let steps 2-3 determine resolution
+        WebIDL::SimpleException exception {
+            WebIDL::SimpleExceptionType::TypeError,
+            "Battery Status API is not yet implemented"sv
+        };
+        m_battery_promise = WebIDL::create_rejected_promise_from_exception(realm, move(exception));
+    }
+
+    // 2. If this's relevant global object's associated Document is not allowed to use the "battery"
+    //    policy-controlled feature, then reject this.[[BatteryPromise]] with a "NotAllowedError" DOMException.
+    // FIXME: Check if Document is allowed to use "battery" policy-controlled feature
+
+    // 3. Otherwise:
+    //    1. If this.[[BatteryManager]] is null, then set it to the result of creating a new BatteryManager
+    //       in this's relevant realm.
+    // FIXME: Create new BatteryManager in this's relevant realm
+
+    //    2. Resolve this.[[BatteryPromise]] with this.[[BatteryManager]].
+    // FIXME
```

Comment:
Instead of adding new `FIXME` lines, you can just `FIXME` the spec steps themselves. E.g.:

```cpp
    // FIXME: 1. If this.[[BatteryManager]] is null, then set it to the result of creating a new BatteryManager
    //    in this's relevant realm.
```